app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuHomepageRobot.kt

```diff
@@ -1,23 +1,64 @@
 package org.mozilla.fenix.ui.robots
 
 import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers
 import androidx.test.espresso.matcher.ViewMatchers.hasSibling
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
 import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.uiautomator.UiSelector
 import org.hamcrest.CoreMatchers.allOf
+import org.junit.Assert.assertTrue
 import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
 import org.mozilla.fenix.helpers.click
 
 /**
  * Implementation of Robot Pattern for the settings Homepage sub menu.
  */
 class SettingsSubMenuHomepageRobot {
 
+    fun verifyHomePageView() {
+        assertMostVisitedTopSitesButton()
+        assertJumpBackInButton()
+        assertRecentBookmarksButton()
+        assertRecentSearchesButton()
+        assertPocketButton()
+        assertOpeningScreenHeading()
+        assertHomepageButton()
+        assertLastTabButton()
+        assertHomepageAfterFourHoursButton()
+    }
+
     fun clickStartOnHomepageButton() = homepageButton().click()
 
-    class Transition
+    class Transition {
+
+        fun goBack(interact: HomeScreenRobot.() -> Unit): HomeScreenRobot.Transition {
+            goBackButton().click()
+
+            HomeScreenRobot().interact()
+            return HomeScreenRobot.Transition()
+        }
+    }
 }
 
+private fun mostVisitedTopSitesButton() =
+    mDevice.findObject(UiSelector().textContains("Most visited top sites"))
+
+private fun jumpBackInButton() =
+    mDevice.findObject(UiSelector().textContains("Jump back in"))
+
+private fun recentBookmarksButton() =
+    mDevice.findObject(UiSelector().textContains("Recent bookmarks"))
+
+private fun recentSearchesButton() =
+    mDevice.findObject(UiSelector().textContains("Recent searches"))
+
+private fun pocketButton() =
+    mDevice.findObject(UiSelector().textContains("Pocket"))
+
```

Comment:
Any resource ID's we can use for these ^^^^^

Reply:
Please see my comment.